projects/plugins/jetpack/tests/e2e/lib/env/playwright-environment.js

```diff
@@ -0,0 +1,155 @@
+const PlaywrightEnvironment = require( 'jest-playwright-preset/lib/PlaywrightEnvironment' ).default;
+const fs = require( 'fs' );
+const logger = require( '../logger' ).default;
+
+class PlaywrightCustomEnvironment extends PlaywrightEnvironment {
+	async setup() {
+		await super.setup();
+	}
+
+	async teardown() {
+		await super.teardown();
+	}
+
+	async handleTestEvent( event ) {
+		await super.handleTestEvent( event );
+
+		let testName = 'unknown_test';
+		let hookName = 'unknown_hook';
+
+		if ( event.test ) {
+			testName = `${ event.test.parent.name } - ${ event.test.name }`;
+		}
+
+		if ( event.hook ) {
+			hookName = `${ event.hook.type } - ${ event.hook.parent.name }`;
+		}
+
+		switch ( event.name ) {
+			case 'setup':
+				break;
+			case 'add_hook':
+				break;
+			case 'add_test':
+				break;
+			case 'run_start':
+				break;
+			case 'test_skip':
+				break;
+			case 'test_todo':
+				break;
+			case 'start_describe_definition':
+				break;
+			case 'finish_describe_definition':
+				break;
+			case 'run_describe_start':
+				break;
+			case 'test_start':
+				logger.info( `START TEST: ${ testName }` );
+				// await this.storeVideoFileName( testName );
+				break;
+			case 'hook_start':
+				logger.info( `START: ${ hookName }` );
+				// await this.storeVideoFileName( hookName );
+				break;
+			case 'hook_success':
+				logger.info( `SUCCESS: ${ hookName }` );
+				break;
+			case 'hook_failure':
+				logger.info( `FAILED: ${ hookName }` );
+				logger.error( JSON.stringify( event.hook.errors ) );
+				logger.error( JSON.stringify( event.error ) );
+				await this.saveScreenshot( hookName );
+				await this.storeVideoFileName( hookName );
+				await this.logHTML( hookName );
+				await this.logFailureToSlack( event.hook.parent.name, event.hook.type, event.error );
+				break;
+			case 'test_fn_start':
+				break;
+			case 'test_fn_success':
+				logger.info( `TEST PASSED: ${ testName }` );
+				break;
+			case 'test_fn_failure':
```

Comment:
can we also capture debuglog on any failure (also in hook failure)

Reply:
I missed that one! 
At a glance it will probably cause some pain because of the playwright-environment not being transformed, but I'll try.

Reply:
Done now. I also moved `logDebugLog()` from `page-helper` to `utils-helper`, it looks as a more appropriate place for it.